prog/dftb+/api/mm/mmapi.F90

```diff
@@ -467,6 +473,64 @@ subroutine convertAtomTypesToSpecies(typeNumbers, species, speciesNames, typeNam
 
   end subroutine convertAtomTypesToSpecies
 
+  !TODO(Alex) Consider whether this should be a member function or free function
+  !> Checks speciesNames has not changed between calls to DFTB+
+  subroutine TDftbPlus_checkSpeciesNames(this, inputSpeciesNames)
+    !TODO(Alex) Establish how to set len of inputSpeciesNames
+    !use dftbp_accuracy,    only: mc
+    use dftbp_message, only: error
+    class(TDftbPlus),               intent(inout) :: this
+    character(len=3),  allocatable, intent(in)    :: inputSpeciesNames(:)
+    logical                                       :: tSpeciesNameChanged
+
+    call this%checkInit()
+    
+    tSpeciesNameChanged = checkSpeciesNames(inputSpeciesNames)
+    
+    if(tSpeciesNameChanged)then
+       call error('speciesNames has changed between calls to DFTB+. This will cause erroneous results.')
+    else
+       continue
+    endif
+
+  end subroutine TDftbPlus_checkSpeciesNames
+
+  !> Set species and all variables/data dependent on it   
+  subroutine TDftbPlus_setSpeciesAndDependents(this, inputSpeciesNames, inputSpecies)
+    !> Instance
+    class(TDftbPlus), intent(inout) :: this
+    
+    !> Type of each atom (nAllAtom) 
+    integer, allocatable, intent(in) :: inputSpecies(:)
+    
+    !> Labels of atomic species (nSpecies)
+    character(mc), allocatable, intent(in) :: inputSpeciesNames(:)
+    
+    call this%checkInit()
+    !TODO(Alex) May not want this call here 
+    call this%checkSpeciesNames(inputSpeciesNames)
```

Comment:
Check speciesNames hasn't changed here or leave to the developer to call where appropriate?

Reply:
Would the API handle in its current status cases, where the specie names had been changed? I think, rather not, so let's check for it explicitely.